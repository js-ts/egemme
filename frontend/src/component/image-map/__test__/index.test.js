/*
 * @Author: qiuz
 * @Github: <https://github.com/qiuziz>
 * @Date: 2019-11-28 10:30:01
 * @Last Modified by: qiuz
 * @Last Modified time: 2019-11-28 13:30:59
 */

import * as React from 'react';
import { render, mount } from 'enzyme';
import { ImageMap } from '../index';
import EXAMPLE from '../../images/example.png';

describe('ImagePicker', () => {
  it('renders correctly', () => {
    const renderDom = render(
      <ImageMap />,
    );

    expect(renderDom).toMatchSnapshot();
	});
	
  it('renders className correctly', () => {
    const wrapper = mount(
			<ImageMap
				className="image-map-test"
				src={EXAMPLE}
      />,
		);
		
    expect(wrapper.find(".image-map-test").exists());
	});

  it('renders img correctly', () => {
    const wrapper = mount(
      <ImageMap
				src={EXAMPLE}
      />,
		);
		
    // show img
    expect(wrapper.find("img").prop("src")).toEqual(EXAMPLE);

	});
	
	it('render map correctly', () => {
		const wrapper = mount(
      <ImageMap
				src={EXAMPLE}
      />,
		);

		expect(wrapper.find("span").length).toEqual(0);
		
		const mapArea = [{"left":"0%","top":"6%","height":"12%","width":"33%"}];
		const wrapperMap = mount(
      <ImageMap
				src={EXAMPLE}
				map={mapArea}
      />,
		);

		expect(wrapperMap.find("span").length).toEqual(1);
	});

	it('Test click event', () => {
		const mockCallBack = jest.fn(index => index);
		const mapArea = [{"left":"0%","top":"6%","height":"12%","width":"33%"}];
		const wrapperMapClick = mount(
      <ImageMap
				src={EXAMPLE}
				map={mapArea}
				onMapClick={mockCallBack}
      />,
		);
		const map1 = wrapperMapClick.find('span').at(0);

		expect(map1.exists());

		map1.simulate('click');
		
		expect(mockCallBack).toBeCalledWith(0);
  });

});
