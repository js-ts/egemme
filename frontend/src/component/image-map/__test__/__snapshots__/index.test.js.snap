// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker Test click event 1`] = `ReactWrapper {}`;

exports[`ImagePicker renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "image-map__content ",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "alt": "",
          "src": "",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "alt": undefined,
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "alt": undefined,
          "src": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
